calico_containers/adapter/datastore.py

```diff
@@ -590,6 +596,87 @@ def remove_ip_pool(self, version, pool):
             raise KeyError("%s is not a configured IP pool." % pool)
 
     @handle_errors
+    def get_hostnames(self):
+        """
+        Return a list of hostnames
+        :return:  A list of hostnames configured in etcd.
+        """
+        # Get a list of the hosts.
+        hosts = []
+        results = self.etcd_client.read(HOSTS_PATH)
+        for result in results.children:
+            packed = result.key.split("/")
+            # Handle no hosts being configured (etcd returns the parent).
+            if len(packed) != 5:
+                continue
+            hosts.append(packed[-1])
+        return hosts
+
+    @handle_errors
+    def get_node_as_peer(self, version, hostname):
```

Comment:
Yup that seems reasonable.

How about  _node_to_bgppeer or _bgppeer_from_node etc.